### ISSUE TITLE
[$1000 in SQT] Query chain state at the current height

### ISSUE THREAD

**Issue by stwiname** [subquery/subql-cosmos#241]
It is not always practical or possible to get all the information required from handlers and the only way to get that information is by querying the contract state. Subql-cosmos allows querying contract state through the injected client but this will return the current state and not the state at the height being indexed. 

This feature should inject a client to the sandbox that limits querying state to the current height being indexed.

### Other requirements
- [Documentation](https://github.com/subquery/documentation) should be updated to include the feature
- Tests with examples that query the correct data at a specific height

Here is an example of querying wasm state at a specific height:
https://github.com/aelesbao/subql-cosmos/commit/c71a4db406d3580a7d529122c0ab17b6ac181eed

**Comment by jilt** [comment]
So this is for Cosmos?
Shouldn't be that hard with their contract API...
I worked on Mars protocol recently, may be able to come up with a solution

**Comment by jamesbayly** [comment]
@jilt that is correct, we'd appreciate some pointers here if you know exactly how to do it.

**Comment by pbkompasz** [comment]
@jamesbayly Hi. I'd like to give this a shot. Is the bounty still valid?
